From 3430da64c2f5ba13be59874a91777ff12bab42e9 Mon Sep 17 00:00:00 2001
Message-ID: <3430da64c2f5ba13be59874a91777ff12bab42e9.1732837202.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1732837202.git.sharlatanus@gmail.com>
References: <cover.1732837202.git.sharlatanus@gmail.com>
From: "Artyom V. Poptsov" <poptsov.artyom@gmail.com>
Date: Sun, 24 Nov 2024 13:27:29 +0300
Subject: [PATCH 09/15] gnu: Add go-github-com-charmbracelet-lipgloss.

* gnu/packages/golang-xyz.scm (go-github-com-charmbracelet-lipgloss): New variable.

Change-Id: I73050c96830025ed63be7567873f8faf7573453d
Reviewed-by: Sharlatan Hellseher <sharlatanus@gmail.com>
---
 gnu/packages/golang-xyz.scm | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/gnu/packages/golang-xyz.scm b/gnu/packages/golang-xyz.scm
index 81dc15ae5d..7abfe589cc 100644
--- a/gnu/packages/golang-xyz.scm
+++ b/gnu/packages/golang-xyz.scm
@@ -1559,6 +1559,35 @@ (define-public go-github-com-cheggaaa-pb-v3
      (modify-inputs (package-propagated-inputs go-github-com-cheggaaa-pb)
        (append go-github-com-vividcortex-ewma)))))
 
+(define-public go-github-com-charmbracelet-lipgloss
+  (package
+    (name "go-github-com-charmbracelet-lipgloss")
+    (version "1.0.0")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/charmbracelet/lipgloss")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "0d1aqzsjy0mcliydbfbg223xxpf9646frbj35ac4fisdy3w3n142"))))
+    (build-system go-build-system)
+    (arguments
+     (list
+      #:import-path "github.com/charmbracelet/lipgloss"))
+    (propagated-inputs
+     (list go-github-com-aymanbagabas-go-udiff
+           go-github-com-charmbracelet-x-ansi
+           go-github-com-charmbracelet-x-exp-golden
+           go-github-com-muesli-termenv
+           go-github-com-rivo-uniseg))
+    (home-page "https://github.com/charmbracelet/lipgloss")
+    (synopsis "Style definitions for nice terminal layouts")
+    (description
+     "Style definitions for nice terminal layouts.  Built with TUIs in mind.")
+    (license license:expat)))
+
 (define-public go-github-com-charmbracelet-x-ansi
   (package
     (name "go-github-com-charmbracelet-x-ansi")
-- 
2.46.0

